src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -1043,6 +1044,58 @@ def print_fields(fields):
                 quoting=csv.QUOTE_NONE,
             )
 
+def resolve_validate_dx_path(path):
+    """
+    Resolves dx path into project, folder and name. Fails if non existing folder is provided. 
+    """
+    
+    project, folder, name = resolve_path(path)
+    if folder != '/':
+        folder_name = '/{}'.format(os.path.basename(folder))
+        folder_path = os.path.dirname(folder)
+        try:
+            folder_exists = check_folder_exists(project, folder_path, folder_name )
+        except Exception as e:
+            raise ResolutionError(e)
+        
+        if not folder_exists:
+            raise ResolutionError('The folder: {} could not be found in the project: {}'.format(folder, project))
+    
+    return project, folder, name
+
+def create_cohort(args): 
+    """
+    Create a cohort from dataset/cohort and specified list of samples. 
+    """
+    #### Validation ####
+    # validate input combinations
+    if args.cohort_ids and args.cohort_ids_file:
```

Comment:
You could use mutex groups in argparse in dx.py to handle this. See example in extract_assay germline(retreive-allele, retrieve-annotation, retrieve-genotype) and extract_assay somatic

Reply:
Thats very elegant!